File: iosApp/iosApp/presentationLayer/pages/familyList/viewModels/FamilyListViewState.swift
enum FamilyListViewState: Equatable {
    case initial
    case loading
    case success
    case error(
        message: String,
        retryAction: (() -> Void)?
    )

    static func == (lhs: FamilyListViewState, rhs: FamilyListViewState) -> Bool {
        switch (lhs, rhs) {
        case (.initial, .initial):
            return true
        case (.loading, .loading):
            return true
        case (.success, .success):
            return true
        case (.error, .error):
            return true
        default:
            return false
        }
    }
}

File: androidApp/src/main/java/dev/haroldjose/familysharedlist/android/presentationLayer/pages/familyList/viewmodels/FamilyListViewState.kt
package dev.haroldjose.familysharedlist.android.presentationLayer.pages.familyList.viewmodels

sealed class FamilyListViewState {

    object Initial: FamilyListViewState()
    object Loading: FamilyListViewState()
    object Success: FamilyListViewState()
    data class Error(
        val message: String,
        val retryAction: (() -> Unit)? = null
    ) : FamilyListViewState()
}